wagtail/contrib/forms/views.py

```diff
@@ -140,6 +144,9 @@ def get_context_data(self, **kwargs):
                 "submissions": self.submissions,
             }
         )
+        next_url = get_valid_next_url_from_request(self.request)
+        if next_url:
+            context["next_url"] = next_url
```

Comment:
This needs a fallback I think, i.e. if the user comes to this view without a `next_url` parameter then it should have a default instead of rendering the input with an empty value. We should reuse `get_success_url()` instead